Compiler/src/Compiler.cpp

```diff
@@ -1477,6 +1480,70 @@ struct Compiler
         }
     }
 
+    void compileExprInterpString(AstExprInterpString* expr, uint8_t target, bool targetTemp)
+    {
+        std::string formatString;
+
+        size_t stringsLeft = expr->strings.size;
+
+        for (AstArray<char> const& string : expr->strings)
+        {
+            std::string stringEscaped(string.data);
+
+            for (size_t characterIndex = 0; characterIndex < stringEscaped.size(); ++characterIndex)
+            {
+                if (stringEscaped[characterIndex] == '%')
+                {
+                    stringEscaped.insert(characterIndex, 1, '%');
+                    characterIndex++;
+                }
+            }
+
+            formatString += stringEscaped;
+
+            stringsLeft--;
+
+            if (stringsLeft > 0)
+                formatString += "%*";
+        }
+
+        auto formatStringSize = formatString.size();
```

Comment:
Please don't use auto in general
```suggestion
        size_t formatStringSize = formatString.size();
```